snorkeling_helper/database_helper/database_modules.py

```diff
@@ -0,0 +1,342 @@
+                            )
+
+                            writer.writerow(
+                                {
+                                    "document_id": document_id,
+                                    "section": section.lower(),
+                                    "position": position
+                                    + section_counter[
+                                        section
+                                    ],  # sections are broken by paragraphs and titles
+                                    "text": sentence.text,
+                                    "word": "|".join(word),
+                                    "pos_tag": "|".join(pos_tag),
+                                    "lemma": "|".join(lemma),
+                                    "dep": "|".join(dep),
+                                    "char_offset": "|".join(
+                                        char_offset
+                                    ),  # act like title and abstract are one string
+                                }
+                            )
+
+                        # Update the section counter based on sections
+                        section_counter.update({section: len(doc_sentences)})
+
+                    del section_counter
+                    del nlp
+
+            data_queue.task_done()
+
+            # Remove subelements from document
+            # Thereby freeing memory
+            clear_document(document)
+            del document
+
+        except Exception as e:
+            print(f"This was an exception {e}")
+            # Do nothing but tell the queue that the task is done
+            data_queue.task_done()
+
+    data_queue.task_done()
+
+
+def supply_documents(doc_type, batch_iterator, data_queue):
+    """
+    This function supples the parse_documents function with xml documents to be parsed.
+
+    Args:
+       - doc_type - specify full text or abstracts
+       - batch_iterator - a generated to read in the documents
+       - data_queue - the queue that holds documents in xml form to be parsed (multiprocessing)
+    """
+    if doc_type == "full":
+        for batch_file in tqdm.tqdm(batch_iterator):
+            tree = ET.parse(str(batch_file.resolve()))
+            root = tree.getroot()
+
+            for document in root.xpath("document"):
+                try:
+                    data_queue.put(ET.tostring(document))
+                except Exception as e:
+                    print(e)
+                    pass
+                finally:
+                    root.clear()
+                    del tree
+
+    elif doc_type == "abstract":
+        for event, document in tqdm.tqdm(batch_iterator):
+            try:
+                data_queue.put(ET.tostring(document))
+            except Exception as e:
+                print(e)
+                pass
+            finally:
+                document.clear()
+    else:
+        raise ("Please provide either full or abstract for doc type")
```

Comment:
You can be more explicit here by raising a ValueError (also I think that using a tuple with `raise` will raise a TypeError because `raise` expects an Exception)